### PR TITLE
LPS-121855 Migrate v2 usages in account/account-admin-web

### PR DESCRIPTION
**Pull request by jonmak08**
Hey @liferay-frontend,

Attached is an update for http://issues.liferay.com/browse/LPS-121855.

@markocikos, @julien, and @ambrinchaudhary,

This one is more just a question regarding https://github.com/liferay-frontend/liferay-portal/pull/775/commits/c45978e3c8adb15122f1d3ed140fd3b36115e886 to see if there is any feedback on those changes.

**Test Plan:**

1. Login to Liferay
2. Click on the **Application Menu**
3. Click on **Accounts**
4. Add an account
5. Click on the **Application Menu**
6. Click on **Account Users**
7. Click on the `+` to select an account user
8. Click on **Choose**
9. Create a new user

admin-users-account

Please let me know if you have any questions. Thanks!

### CODE REVIEW

**Comment by jonmak08** [comment]
ci:stop

**Comment by jonmak08** [comment]
ci:test:sf

**Comment by jonmak08** [comment]
ci:test:relevant

**Comment by jonmak08** [comment]
ci:stop

**Comment by jonmak08** [comment]
ci:test:sf

**Comment by jonmak08** [comment]
ci:stop

**Comment by jonmak08** [comment]
ci:test:sf

**Comment by jonmak08** [comment]
ci:test:relevant

**Comment by jonmak08** [comment]
ci:test:sf

**Comment by jonmak08** [comment]
ci:test:relevant

File: modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_users_admin/js/AccountUsersAdminManagementToolbarPropsTransformer.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+import {
+	createPortletURL,
+	navigate,
+	openSelectionModal,
+	postForm,
+} from 'frontend-js-web';
+
+const _updateAccountUsers = (portletNamespace, url) => {
+	const form = document.getElementById(`${portletNamespace}fm`);
```
Comment (julien) [inline modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_users_admin/js/AccountUsersAdminManagementToolbarPropsTransformer.js:23]: Non blocking, you don't have to name your function with a leading underscore

File: modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_users_admin/js/AccountUsersAdminManagementToolbarPropsTransformer.js [hunk 2]
```diff
@@ -13,87 +13,67 @@
...
+	const activateAccountUsers = (itemData) => {
+		_updateAccountUsers(itemData.activateAccountUsersURL);
+	};
 
-	deactivateAccountUsers(itemData) {
+	const addAccountUser = (itemData) => {
+		openSelectionModal({
+			id: `${portletNamespace}addAccountUser`,
+			onSelect: (selectedItem) => {
+				var addAccountUserURL = createPortletURL(
```
Comment (markocikos) [inline modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_users_admin/js/AccountUsersAdminManagementToolbarPropsTransformer.js]: We should use `const`. There is one more `var` in file.

File: modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_users_admin/js/AccountUsersAdminManagementToolbarPropsTransformer.js [hunk 3]
```diff
@@ -106,21 +86,73 @@ class ManagementToolbarDefaultEventHandler extends DefaultEventHandler {
...
-	_updateAccountUsers(url) {
-		const form = this.one('#fm');
+	const selectAccountEntries = (itemData) => {
+		_openAccountEntrySelector(
+			Liferay.Language.get('select'),
+			`${portletNamespace}selectAccountEntries`,
+			Liferay.Language.get(itemData.dialogTitle),
+			itemData.accountEntriesSelectorURL,
+			(selectedItems) => {
+				var redirectURL = Liferay.Util.PortletURL.createPortletURL(
```
Comment (markocikos) [inline modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_users_admin/js/AccountUsersAdminManagementToolbarPropsTransformer.js]: We can use `createPortletURL`, like in `addAccountUser`. It is already imported.

File: modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_users_admin/js/AccountUsersAdminManagementToolbarPropsTransformer.js [hunk 4]
```diff
@@ -106,21 +86,73 @@ class ManagementToolbarDefaultEventHandler extends DefaultEventHandler {
...
+					itemData.redirectURL,
+					{
+						accountEntriesNavigation: 'accounts',
+						accountEntryIds: selectedItems.value,
+					}
+				);
+
+				window.location.href = redirectURL;
```
Comment (markocikos) [inline modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_users_admin/js/AccountUsersAdminManagementToolbarPropsTransformer.js]: We can use `navigate`, like in `addAccountUser`. It is already imported.

File: modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_users_admin/js/AccountUsersAdminManagementToolbarPropsTransformer.js [hunk 5]
```diff
@@ -106,21 +86,73 @@ class ManagementToolbarDefaultEventHandler extends DefaultEventHandler {
...
+		);
+	};
 
-		Liferay.Util.postForm(form, {
+	const _updateAccountUsers = (url) => {
+		const form = document.getElementById(`${portletNamespace}fm`);
+
+		if (!form) {
+			return;
+		}
```
Comment (markocikos) [inline modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_users_admin/js/AccountUsersAdminManagementToolbarPropsTransformer.js]: Empty line typo.

**Review by markocikos — COMMENTED** [review]
Few minor nits in comment, otherwise LGTM!

File: modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_users_admin/js/AccountUsersAdminManagementToolbarPropsTransformer.js (hunk 5 above)
Comment (julien) [reply modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_users_admin/js/AccountUsersAdminManagementToolbarPropsTransformer.js]: @markocikos I was just curious, what's an empty line typo?

Comment (markocikos) [reply modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_users_admin/js/AccountUsersAdminManagementToolbarPropsTransformer.js]: There is an empty line missing here 
```js
}
postForm(form, {
```
 I am assuming it's a typo :slightly_smiling_face:

Comment (julien) [reply modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_users_admin/js/AccountUsersAdminManagementToolbarPropsTransformer.js]: Oh ok, :+1:  understood thanks.

**Comment by jonmak08** [comment]
ci:close

**Comment by jonmak08** [comment]
New Pull Request: https://github.com/jonmak08/liferay-portal/pull/2124

**Pull request closed without merging**